Sources/StreamChat/Workers/MessageUpdater.swift

```diff
@@ -60,13 +60,19 @@ class MessageUpdater: Worker {
                 // to try to delete the message on the backend.
                 return
             }
-
-            messageDTO.isHardDeleted = hard
             
-            if messageDTO.existsOnlyLocally && !isLocalStorageEnabled {
+            messageDTO.isHardDeleted = hard || messageDTO.isBounced
+            
+            if messageDTO.existsOnlyLocally && (!isLocalStorageEnabled || messageDTO.isBounced) {
```

Comment:
I would update `existsOnlyLocally` to return `true` if the message is bounced and keep this condition unchanged